crd/multitenantpodnetworkconfig/api/v1alpha1/multitenantpodnetworkconfig.go

```diff
@@ -44,6 +45,8 @@ type MultitenantPodNetworkConfigList struct {
 type MultitenantPodNetworkConfigSpec struct {
 	// name of PNI object from requesting cx pod
 	PodNetworkInstance string `json:"podNetworkInstance,omitempty"`
+	// name of PN object from requesting cx pod
+	PodNetwork string `json:"podNetwork,omitempty"`
```

Comment:
should one of the PN or PNI be optional?

Reply:
PNI is optional.

Reply:
Doesn't omitempty means that field is optional? Is yes, should I remove omitempty from PN here?

Reply:
not exactly. `omitempty` means if the field is empty/zero-valued, that it will not even be included in the JSON representation of the object. If we had a `size int` field, if the value was `0` then `size: 0` would not even be included in the JSON.
there is a [kubebuilder annotation](https://github.com/Azure/azure-container-networking/blob/906313d49fad628ccb2eb7efc3d7702639d29a5c/crd/nodenetworkconfig/api/v1alpha/nodenetworkconfig.go#L64-L65) that we should be using to mark optional fields (and maybe we should be explicit about required fields also, but I think the default is required if unspecified).

Reply:
// +kubebuilder:validation:Optional  - This only seems to work when 'omitempty' tag is not present. 
What I mean by aboe is,  without the optional validation annotation and omitempty tag, if I regen the CRD manifest, the field is marked as required 
But if it is tagged with omitempty, adding above annotation has no effect to the generated CRD manifest.

Reply:
So basically if I add this annotation - // +kubebuilder:validation:Optional , it doesn't have any change in generated CRD manifest. Only after omitempty is removed, it makes the field optional.